python-package/lightgbm/basic.py

```diff
@@ -2008,7 +2008,7 @@ def __init_from_list_np2d(
             ptr_data = (ctypes.POINTER(ctypes.c_float) * len(mats))()
 
         holders = []
-        type_ptr_data = None
+        type_ptr_data = -1
```

Comment:
This should always get overwritten in the `for` loop below, given that the only call to this function checks that `mats` has at least one element

https://github.com/microsoft/LightGBM/blob/638014d5c56fb92396bd55f344a47e3f651a3cd4/python-package/lightgbm/basic.py#L1851-L1853

So this initial value, whether it's `None` or `-1`, should never end up actually getting used.